### PR TITLE
stats: exposed and documented

### LINKED ISSUE #168: Expose switch stats
Now that libp2p-switch is about to get [some networking stats](https://github.com/libp2p/js-libp2p-switch/pull/243), we probably should expose and document them in libp2p.

My proposal is simply directly exposing `switch.stats` into `libp2p.stats` and porting over the documentation.

Thoughts?

### PR DESCRIPTION
**Pull request by pgte**
Fixes #168

### CODE REVIEW

File: test/stats.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+const chai = require('chai')
+chai.use(require('dirty-chai'))
+const expect = chai.expect
+
+const createNode = require('./utils/node').createNode
+
+describe('libp2p', (done) => {
+  it('has stats', () => {
+    createNode('/ip4/0.0.0.0/tcp/0', {
```
Comment (VictorBjelkholm) [inline test/stats.js]: Probably want to set address to `127.0.0.1` instead, no reason to bind to public interface :)

Comment (pgte) [reply test/stats.js]: @VictorBjelkholm good call, thanks! Fixed by 10bd08bdd51f1beaf0856d366e6f3f79bbf1a09f.

**Comment by hacdias** [comment]
@pgte just updated the version of `libp2p-switch` on `package.json` to see if the tests work 😄

**Comment by hacdias** [comment]
Any updates here?

**Comment by hacdias** [comment]
The tests seem to be failing on some unrelated to stats code:

https://github.com/libp2p/js-libp2p/blob/7eb116615e3a3ac192a55d9f49649bfc580305be/test/peer-routing.node.js#L52-L60

/cc @pgte

**Comment by diasdavid** [comment]
@dryajov did something change on circuit-relay?

image

**Comment by hacdias** [comment]
Ping @pgte

**Comment by pgte** [comment]
@hacdias my build runs locally, but I'm trying to figure out what's happening on CI..

**Comment by diasdavid** [comment]
@pgte suggestion: try to rebase master onto this branch, it might solve your issue.

**Comment by hacdias** [comment]
@pgte @diasdavid just rebased into master and the issue persists. I'm using the last commit from here on https://github.com/ipfs/js-ipfs/pull/1230 and it seems to be working well.

**Comment by diasdavid** [comment]
This one is stale for a while, what's blocking from getting CI green?

**Comment by pgte** [comment]
Related: https://github.com/libp2p/js-libp2p-switch/pull/246

**Comment by pgte** [comment]
I cannot reproduce this locally, but the error traces suggest that [this PR for libp2p-switch](https://github.com/libp2p/js-libp2p-switch/pull/246) should ameliorate this.

**Comment by diasdavid** [comment]
restarted CI now that libp2p-switch fix has been merged and released as a patch version.

**Comment by pgte** [comment]
@diasdavid thanks! That fixed some issues, but now I have node v6-only failings I'm going to dive into..

**Comment by pgte** [comment]
Update: for some reason, in node.js v6, in some instances, `peerInfo.id` is a buffer and not a `peer-id` instance..

**Comment by pgte** [comment]
I think I found the problem, and it's in libp2p-circuit: https://github.com/libp2p/js-libp2p-circuit/pull/23

If ok, needs updating libp2p-circuit and downstream libp2p-switch and libp2p.

**Comment by diasdavid** [comment]
@pgte a new consistent error as arise: https://travis-ci.org/libp2p/js-libp2p/builds/353334562#L4862

**Comment by pgte** [comment]
One last edge case fixed in the switch observer: https://github.com/libp2p/js-libp2p-switch/pull/248

It should all be ok after this switch fix:

1__bash

**Pull request merged**